taichi/transforms/offload.cpp

```diff
@@ -489,9 +489,8 @@ void insert_gc(IRNode *root) {
     auto snodes = gc_statements[i].second;
     for (auto *snode : snodes) {
       if (is_gc_able(snode->type)) {
-        b->statements.insert(
-            b->statements.begin() + i + 1,
-            Stmt::make<OffloadedStmt>(OffloadedStmt::TaskType::gc, snode));
+        b->insert(Stmt::make<OffloadedStmt>(OffloadedStmt::TaskType::gc, snode),
+                  i + 1);
```

Comment:
Modifying `Block::statements` directly without setting `parent` was really bad...

If we set `Block::statements` to be `private` to prevent code fragments like this, we may need some functions like `Block::get_statements()` to keep code fragments like `for (auto &stmt : b->statements)` neat?

Reply:
Right, probably something like `Block::get_statements() const`.